web platform tests that build up keydown and keyup events using test_driver.Actions fail

rniwa opened the pull request:
#### 790c299b40fb9819ce4511276537ec7cba3abf3c
<pre>
web platform tests that build up keydown and keyup events using test_driver.Actions fail
<a href="https://bugs.webkit.org/show_bug.cgi?id=249460">https://bugs.webkit.org/show_bug.cgi?id=249460</a>

Reviewed by Tim Nguyen.

Added the basic support for test_driver.Actions.keyDown/keyUp.

* LayoutTests/TestExpectations: Added a image-only failing expectation on grid-layout-stale-002.html since this test now fails.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scroll-snap/selection-target-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-button-element/active-onblur-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/checkable-active-onblur-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-canceling-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/inert-node-is-uneditable-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/compat/pointerevent_mouseevent_key_pressed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_touch-action-keyboard-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js:
(SeleniumCharCodeToEventSenderKey): Moved outside the function.
(convertSeleniumKeyCode): Ditto.
(window.test_driver_internal.send_keys):
(window.test_driver_internal.action_sequence): Added the support for key source.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/accesskey.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/delegatesFocus-highlight-sibling-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-fallback-default-tabindex-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-fallback-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-2levels-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-delegatesFocus-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-fallback-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-shadow-in-fallback-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-shadow-in-slot-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-with-tabindex-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slots-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slots-in-slot-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-web-component-input-type-radio-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-with-delegatesFocus-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-nested-slots-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-reverse-unassignable-slot-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-reverse-unassigned-slot-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-unassignable-slot-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-with-negative-index-expected.txt:
* LayoutTests/platform/gtk/TestExpectations: Added failing test expectations.
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/semantics/forms/the-button-element/active-onblur-expected.txt: Added.
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/checkable-active-onblur-expected.txt: Added.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/semantics/forms/the-button-element/active-onblur-expected.txt: Added.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/checkable-active-onblur-expected.txt: Added.

Canonical link: <a href="https://commits.webkit.org/259822@main">https://commits.webkit.org/259822@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3b8727b87fc06de71d40a4f76a2dbfbb2671f258

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/105991 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/15048 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38825 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/115174 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/175285 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16472 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/6233 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/98202 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/114914 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111744 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/12542 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95511 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/40081 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/94407 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/27154 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/81750 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/8304 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28506 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/8792 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/5072 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14418 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48049 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10338 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3651 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js:
```diff
@@ -274,11 +279,69 @@ window.test_driver_internal.action_sequence = async function(sources)
+                break;
+            }
+            case 'pause':
+                // FIXME: Use eventSender.leapForward.
+                throw new Error('testdriver-vendor.js for WebKit does not yet support pause key action');
```
Can we use the `pause` function (basically a setTimeout wrapped inside a Promise) ?

rniwa replied:
So as far as I can tell, there is no imported WPT which uses this pause function so I left it unimplemented. We can obviously implement it, but I'd rather do it when we have a concrete test case to work with.

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js:
```diff
@@ -274,11 +279,69 @@ window.test_driver_internal.action_sequence = async function(sources)
         }
     }
 
+    if (keySource && pointerSource)
+        throw new Error("testdriver-vendor.js for WebKit does not yet support mixing key and pointer sources");
```
There was an attempt at fixing this code to make it work with different types of actions in: https://bugs.webkit.org/show_bug.cgi?id=243272 where we started work on implementing wheel actions.

rniwa replied:
Yeah, we should eventually fix that but for now, this PR will do.

nt1m reviewed (approved):
Seems reasonable as a start, we can always refine later.

rniwa committed:
web platform tests that build up keydown and keyup events using test_driver.Actions fail
https://bugs.webkit.org/show_bug.cgi?id=249460

Reviewed by Tim Nguyen.

Added the basic support for test_driver.Actions.keyDown/keyUp.

* LayoutTests/TestExpectations: Added a image-only failing expectation on grid-layout-stale-002.html since this test now fails.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scroll-snap/selection-target-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-button-element/active-onblur-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/checkable-active-onblur-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-canceling-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/inert-node-is-uneditable-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/compat/pointerevent_mouseevent_key_pressed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_touch-action-keyboard-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js:
(SeleniumCharCodeToEventSenderKey): Moved outside the function.
(convertSeleniumKeyCode): Ditto.
(window.test_driver_internal.send_keys):
(window.test_driver_internal.action_sequence): Added the support for key source.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/accesskey.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/delegatesFocus-highlight-sibling-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-fallback-default-tabindex-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-fallback-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-2levels-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-delegatesFocus-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-fallback-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-shadow-in-fallback-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-shadow-in-slot-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-with-tabindex-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slots-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slots-in-slot-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-web-component-input-type-radio-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-with-delegatesFocus-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-nested-slots-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-reverse-unassignable-slot-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-reverse-unassigned-slot-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-unassignable-slot-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-with-negative-index-expected.txt:
* LayoutTests/platform/gtk/TestExpectations: Added failing test expectations.
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/semantics/forms/the-button-element/active-onblur-expected.txt: Added.
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/checkable-active-onblur-expected.txt: Added.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/semantics/forms/the-button-element/active-onblur-expected.txt: Added.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/checkable-active-onblur-expected.txt: Added.

Canonical link: https://commits.webkit.org/259822@main
